### PR TITLE
issue#09

### PR DESCRIPTION
**Pull request by 1danish159**
# Pull Request Template

## Prequisites

- [x] Did you run lint -cf `filename` on your changes, if you are commiting a .c file?
- [x] Did you run lint -sf `filename` on your changes, if you are commiting a .sh file?
- [x] Does your code compile properly using `make`?
- [x] Did you related issue number in the description?

## Description

Values added in mcause.h

## Technical Debt

No

### CODE REVIEW

File: code/include/env/mcause.h [hunk 1]
```diff
@@ -4,7 +4,38 @@
 #define MCAUSE_EXCEPTION_SHIFT 0
 #define MCAUSE_INTERRUPT_SHIFT 31
 
-#define MCAUSE_EXCEPTION_MASK 0xFFFFFFF1ull
-#define MCAUSE_INTERRUPT_MASK 0x80000000ull
+
+#define MCAUSE_I_Supervisor_software_interrupt 0x00000001ull
```
Comment (mmhus) [inline code/include/env/mcause.h]: change everyone to capitals.

File: code/include/env/mcause.h [hunk 2]
```diff
@@ -4,7 +4,38 @@
...
+#define MCAUSE_I_Virtual_supervisor_software_interrupt 0x00000002ull
```
Comment (mmhus) [inline code/include/env/mcause.h]: Align all values

File: code/include/env/mcause.h [hunk 3]
```diff
@@ -4,7 +4,38 @@
...
+#define MCAUSE_I_Supervisor_guest_external_interrupt 0x00000800ull
+
+
+#define MCAUSE_E_Instruction_address_misaligned 0x00000000ull
+#define MCAUSE_E_Instruction_access_fault 0x00000001ull
+#define MCAUSE_E_Illegal_instruction 0x00000002ull
+#define MCAUSE_E_Breakpoint 0x00000004ull
+#define MCAUSE_E_Load_address_misaligned 0x00000008ull
+#define MCAUSE_E_Load_access_fault 0x00000010ull
+#define MCAUSE_E_Store/AMO_address_misaligned 0x00000020ull
```
Comment (mmhus) [inline code/include/env/mcause.h]: never use `/` or `\` in identifiers, replace `\`with `_`, everywhere.

File: code/include/env/mcause.h [hunk 4]
```diff
@@ -4,7 +4,38 @@
...
+#define MCAUSE_E_Store/AMO_access_fault 0x00000040ull
+#define MCAUSE_E_Environment_call_from_U-mode_or_VU-mode 0x00000080ull
```
Comment (mmhus) [inline code/include/env/mcause.h]: only underscores are legal identifiers, replace `-` with `_` everywhere.

File: code/include/env/mcause.h [hunk 5]
```diff
@@ -4,7 +4,38 @@
 #define MCAUSE_EXCEPTION_SHIFT 0
 #define MCAUSE_INTERRUPT_SHIFT 31
```
Comment (mmhus) [inline code/include/env/mcause.h]: Add a description here, using comments to explain MCAUSE is a register of fixed length (32 bits).

**Review by mmhus — CHANGES_REQUESTED** [review]

**Review by 1danish159 — COMMENTED** [review]
Changes made

File: code/include/env/mcause.h [hunk 6]
```diff
@@ -2,9 +2,40 @@
...
-#define MCAUSE_INTERRUPT_SHIFT 31
+#define MCAUSE_INTERRUPT_SHIFT 31 // MCAUSE is a register of fixed length (32 bits)
+
+
+#define MCAUSE_I_SUPERVISOR_SOFTWARE_INTERRUPT              0x00000001ull
+#define MCAUSE_I_VIRTUAL_SUPERVISOR_SOFTWARE_INTERRUPT      0x00000002ull
+#define MCAUSE_I_MACHINE_SOFTWARE_INTERRUPT                 0x00000004ull
+#define MCAUSE_I_SUPERVISOR_TIMER_INTERRUPT                 0x00000010ull
+#define MCAUSE_I_VIRTUAL_SUPERVISOR_TIMER_INTERRUPT         0x00000020ull
+#define MCAUSE_I_MACHINE_TIMER_INTERRUPT                    0x000000400ull
```
Comment (mmhus) [inline code/include/env/mcause.h:13]: Check the extra zero in this line and fix it.

**Review by mmhus — CHANGES_REQUESTED** [review]

**Comment by 1danish159** [comment]
Updated

**Pull request closed without merging**